addons/mail/static/src/js/systray.js

```diff
@@ -136,16 +121,162 @@ var MessagingMenu = Widget.extend({
     on_click_new_message: function () {
         chat_manager.bus.trigger('open_chat');
     },
-    on_click_channel: function (event) {
-        var channel_id = $(event.currentTarget).data('channel_id');
-        var channel = chat_manager.get_channel(channel_id);
-        if (channel) {
-            chat_manager.open_channel(channel);
+
+    // Handlers
+
+    /**
+     * When a channel is clicked on, we want to open chat/channel window
+     * If channel is inbox then redirect to that record view
+     * If record not linked redirect to Inbox
+     * @private
+     * @param {MouseEvent} event
+     */
+    _onClickChannel: function (event) {
+        var channelID = $(event.currentTarget).data('channel_id');
+        if (channelID == 'channel_inbox') {
+            var messageID = $(event.currentTarget).data('message_id');
+            framework.redirect('mail/view?message_id='+messageID);
```

Comment:
why do you do a redirect? it causes the page to reload, and I do not like that...